### PR TITLE
Support react 18

### PR DESCRIPTION
**Pull request by Arcath**
After updating to React 18 the path to the jsx-runtime has changed which means we needto update the import in mdx-bundler.

Its as simple as dropping the `.js` from the import.

I've also updated the tests to be compatible with the new errors from esbuild.

This might technically be a breaking change as it will probabbly break react 17.

- [x] Documentation N/A
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

**Comment by Arcath** [comment]
This overlaps with #162 and should close #164

**Comment by Arcath** [comment]
Here's a question.

With GH actions can we setup a matrix to run against multiple React versions?

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     if: ${{ !contains(github.head_ref, 'all-contributors') }}
     strategy:
       matrix:
-        node: [12, 14]
+        node: [12, 14, 16]
```
Comment (Arcath) [inline .github/workflows/validate.yml:19]: Tests now pass on Node 16 🎉

File: src/__tests__/index.js [hunk 2]
```diff
@@ -195,7 +195,7 @@ import Demo from './demo.blah'
 
   assert.match(
     error.message,
-    `error: [plugin: inMemory] Invalid loader: "blah" (valid: js, jsx, ts, tsx, css, json, text, base64, dataurl, file, binary)`,
+    `ERROR: [plugin: inMemory] Invalid loader value: "blah"`,
```
Comment (Arcath) [inline src/__tests__/index.js:198]: I figuered since esbuild might add other loaders by default so we don't want this test to fail when that happens.

**Review by kentcdodds — APPROVED** [review]
Looks great to me! 👏👏

**Comment by kentcdodds** [comment]
> With GH actions can we setup a matrix to run against multiple React versions?

Yes, it's possible. No, it's not worth it 😅

**Comment by kentcdodds** [comment]
Oh, and I would be quite surprised if this broke React 17 🤔

**Pull request merged**